aptos-move/aptos-vm/src/move_vm_ext/aggregator_extension.rs

```diff
@@ -0,0 +1,648 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::{
+    delta_change_set::{addition, deserialize, serialize, subtraction, DeltaOp},
+    vm_status::StatusCode,
+};
+use better_any::{Tid, TidAble};
+use move_deps::{
+    move_binary_format::errors::{PartialVMError, PartialVMResult},
+    move_core_types::{account_address::AccountAddress, gas_schedule::GasCost},
+    move_table_extension::{TableHandle, TableResolver},
+    move_vm_runtime::{
+        native_functions,
+        native_functions::{NativeContext, NativeFunctionTable},
+    },
+    move_vm_types::{
+        loaded_data::runtime_types::Type,
+        natives::function::NativeResult,
+        pop_arg,
+        values::{IntegerValue, Reference, Struct, StructRef, Value},
+    },
+};
+use smallvec::smallvec;
+use std::{
+    cell::RefCell,
+    collections::{btree_map::Entry, BTreeMap, BTreeSet, VecDeque},
+};
+use tiny_keccak::{Hasher, Sha3};
+
+/// State of an aggregator: `Data` means that aggregator stores an exact value,
+/// `PositiveDelta` means that actual value is not known but must be added later
+#[derive(Clone, Copy, PartialEq)]
+pub enum AggregatorState {
+    Data,
+    PositiveDelta,
```

Comment:
No, we have `state` and `value` separately to avoid annoying and ugly matches, e.g. when the value changes but the state does not. This way we can update the value, and conditionally update the state